js/lib/AlphabeticIndex.js

```diff
@@ -221,128 +353,303 @@ AlphabeticIndex.prototype.addLabels = function(labels, start) {
+    		}
+		}
+	}
+
+	if (collationValue) {
+		if (typeof collationValue[0] === 'number') {
+			if (collationValue[0] < this.flowBoundaries[0]) {
+				label = this.underflowLabel;
+			} else if (collationValue[0] > this.flowBoundaries[1]){
+				label = this.overflowLabel;
+			} else {
+				label = this._getKeyByValue(collationValue);
+			}	
+		} else if (typeof collationValue[0] === 'object') {
+			label = this._getKeyByValue(collationValue[0]);
+		} 
+	} else {
+		charNum = IString.toCodePoint(firstChar, 0);
+		firstBoundaryChar = this._getKeyByValue([this.flowBoundaries[0]]);
+		endBoundaryChar = this._getKeyByValue([this.flowBoundaries[1]]);
+
+		firstCharNum = IString.toCodePoint(firstBoundaryChar, 0);
+		endCharNum = IString.toCodePoint(endBoundaryChar, 0);
+
+		if (charNum < firstCharNum) {
+			label = this.underflowLabel;
+		} else if (charNum > endCharNum) {
+			label = this.overflowLabel;
+		} 
+	}
+
+	return label;
+};
+
+
+/**
+ * Return default indexing style in the current locale.
+ * @returns {string} the default indexing style for this locale.
+ */
+AlphabeticIndex.prototype.getIndexStyle = function() {
+	return this.collation["default"];
 };
 
 /**
  * Return the total number of buckets in this index.
- * 
  * @returns {number} the number of buckets in this index
  */
 AlphabeticIndex.prototype.getBucketCount = function() {
+	var count = Object.keys(this.index).length;
+	return count;
 };
 
 /**
  * Return the bucket labels for this index in order. This
- * will include the under-, in-, and overflow labels if
+ * method only returns the index labels for buckets that
+ * actually contain elements. This
+ * will include the under- and overflow labels if
  * they are used in this index.
- * 
+ *
  * @returns {Array.<String>} the array of bucket labels
  * for this index in collation order
  */
 AlphabeticIndex.prototype.getBucketLabels = function() {
+	var buckets = this.getAllBuckets();
+	var label = new Array();
+	var i;
+
+	for (i=0; i < buckets.length; i++) {
+		label.push(buckets[i].label);
+	}
+	return label;
+};
+
+/**
+ * Return the all the bucket labels typically used in the
+ * locale. This includes all bucket labels, even if those
+ * buckets do not contain any elements.
+ *
+ * @returns {Array.<String>} the array of bucket labels
+ * for this index in collation order
+ */
+AlphabeticIndex.prototype.getAllBucketLabels = function() {
+	var label, i;
+
+	if (this.allBucketLabels) {
+		return this.allBucketLabels;
+	}
+
+	this.allBucketLabels = new Array();	
+	this.allBucketLabels.push(this.underflowLabel);
+
+	for (i=0; i < this.indexUnits.length; i++) {
+		this.allBucketLabels.push(this.indexUnits[i]);
+	}
+
+	this.allBucketLabels.push(this.overflowLabel);
+	return this.allBucketLabels;
 };
 
 /**
  * Return the collator used to sort elements in this
  * index.
- * 
+ *
  * @return {Collator} the ilib Collator instance used
  * in this index
  */
 AlphabeticIndex.prototype.getCollator = function() {
+	return this.collation;
```

Comment:
This is not of type Collator. This instance variable is initialized to the collation.json in the constructor. This should return a real Collator instance.